pynicotine/slskmessages.py

```diff
@@ -325,27 +337,25 @@ def getObject(self, message, type, start=0, getintasshort=0, getsignedint=0, pri
                 return struct.calcsize("<Q")+start, struct.unpack("<Q", message[start:start+struct.calcsize("<Q")])[0]
             else:
                 return start, None
-        except struct.error, error:
+        except struct.error as error:
             if printerror:
                 log.addwarning("%s %s trying to unpack %s at '%s' at %s/%s" % (self.__class__, error, type, message[start:].__repr__(), start, len(message)))
-                self.debug(message)
-            raise struct.error, error
+            raise struct.error(error)
             # return start, None
 
     def packObject(self, object):
         """ Returns object (integer, long or string packed into a
         binary array."""
-        if type(object) is types.IntType:
+        if type(object) is int:
             return struct.pack("<i", object)
-        elif type(object) is types.LongType:
+        elif type(object) is int:
```

Comment:
I'm using `object.bit_length()` here to determine which should be used. Potentially there are problems with that approach since there can be `long` values which occupy less than 33 bit, but I don't see how else this can be done.